iceoryx_hoofs/test/moduletests/test_cxx_newtype.cpp

```diff
@@ -303,4 +303,104 @@ TEST(NewType, CreatingNewTypeWithMacroWorks)
     EXPECT_FALSE(a == b);
 }
 
+TEST(NewType, NewTypeIsIncrementable)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6d03b24b-fc72-409b-aa2a-f19228ff152c");
+    using SutType = Sut<cxx::NewType<uint64_t,
+                                     newtype::Incrementable,
+                                     newtype::MoveConstructable,
+                                     newtype::CopyConstructable,
+                                     newtype::Comparable,
+                                     newtype::ConstructByValueCopy>>;
+    constexpr uint64_t START_VALUE{42};
+    SutType a{START_VALUE};
+    SutType pre{++a};
+    SutType post{a++};
```

Comment:
Why does this also work when I comment out l. 310?

Reply:
@FerdinandSpitzschnueffler That's due to the increment being a free function and indeed defeats the purpose. I'll think of a way how to refactor that.

Reply:
@FerdinandSpitzschnueffler Done :heavy_check_mark: